### PR TITLE
More enum samples

### PR DESCRIPTION
**Pull request by lanmonster**
main (424c891)
```
[ FAIL ] samples/enums/recursive_enums.jakt
[ PASS ] samples/enums/one_line_enum_declaration.jakt
[ FAIL ] samples/enums/parse.jakt
[ FAIL ] samples/enums/is_variant.jakt
[ PASS ] samples/enums/empty_enum.jakt
[ FAIL ] samples/enums/recursive_enums_bad.jakt
[ FAIL ] samples/enums/methods.jakt
[ PASS ] samples/enums/underlying.jakt
[ FAIL ] samples/enums/simple_match.jakt
[ FAIL ] samples/enums/simple_constructor.jakt
[ FAIL ] samples/enums/methods_access_violation.jakt
```

after these commits
```
[ FAIL ] samples/enums/recursive_enums.jakt
[ PASS ] samples/enums/one_line_enum_declaration.jakt
[ PASS ] samples/enums/parse.jakt
[ FAIL ] samples/enums/is_variant.jakt
[ PASS ] samples/enums/empty_enum.jakt
[ PASS ] samples/enums/recursive_enums_bad.jakt
[ FAIL ] samples/enums/methods.jakt
[ PASS ] samples/enums/underlying.jakt
[ FAIL ] samples/enums/simple_match.jakt
[ FAIL ] samples/enums/simple_constructor.jakt
[ FAIL ] samples/enums/methods_access_violation.jakt
```

### CODE REVIEW

**Comment by jntrnr** [comment]
@lanmonster - I'm not sure what the progress part means. Are you saying you're causing some to fail that used to pass and some pass that used to fail?

File: selfhost/typechecker.jakt [hunk 1]
```diff
@@ -4044,6 +4044,74 @@ struct Typechecker {
...
                                         .add_var_to_scope(scope_id: new_scope_id, name: variant_argument.binding, var_id, span)
                                     }
+                                    StructLike(name, fields) => {
+                                        covered_variants.push(name)
+
+                                        mut field_variables: [CheckedVariable] = []
+                                        for var_id in fields.iterator() {
+                                            field_variables.push(.program.get_variable(var_id))
+                                        }
+                                        mut seen_names: [String] = [] // FIXME: HashSet
```
Comment (alimpfard) [inline selfhost/typechecker.jakt:4054]: Any reason not to use `mut seen_names: {String} = {}`?

Comment (lanmonster) [reply selfhost/typechecker.jakt:4054]: Did not know about that!

**Comment by lanmonster** [comment]
> @lanmonster - I'm not sure what the progress part means. Are you saying you're causing some to fail that used to pass and some pass that used to fail?

No, it is just a snapshot of what now passes as a result of these commits. Some pass that used up fail, namely `recursive_enums_bad`

**Pull request merged**